### PR TITLE
RAC-1048: Rename STARTED status + Restrict to 50 pages when has no filter set

### PR DESCRIPTION
**Pull request by ghoussard**
**Description (for Contributor and Core Developer)**

**Rename STARTED status**

Screenshot from 2021-12-03 12-18-32
Screenshot from 2021-12-03 12-18-39

**Restrict to 50 pages**

Screenshot from 2021-12-03 11-59-07

**Definition Of Done (for Core Developer only)**

- [ ] Tests
- [ ] Migration & Installer
- [ ] PM Validation (Story)
- [ ] Changelog (maintenance bug fixes)
- [ ] Tech Doc

### CODE REVIEW

File: src/Akeneo/Platform/Job/back/Infrastructure/Query/SearchJobExecution.php [hunk 1]
```diff
@@ -72,6 +72,9 @@ public function search(SearchJobExecutionQuery $query): array
         $queryParamsTypes = $this->buildQueryParamsTypes();
 
         $page = $query->page;
+        if(!$query->hasOneFilterSet() && $page > 50) {
```
Comment (ghoussard) [inline src/Akeneo/Platform/Job/back/Infrastructure/Query/SearchJobExecution.php:75]: This validation line will be moved in controller in another PR (https://akeneo.atlassian.net/browse/RAC-1101?atlOrigin=eyJpIjoiYjk1OTQ1NjdhNGM3NDM3NTg2YzI5Y2ZiNmEyY2VlYmIiLCJwIjoiaiJ9)

File: src/Akeneo/Platform/Job/back/tests/.php_cd.php [hunk 2]
```diff
@@ -27,7 +27,6 @@
             'Akeneo\Platform\Bundle\InstallerBundle\Event\InstallerEvents',
             'Akeneo\Platform\Job\Application',
             'Akeneo\Platform\Job\Domain',
-            'Akeneo\Tool',
```
Comment (ValentinMumble) [inline src/Akeneo/Platform/Job/back/tests/.php_cd.php:30]: 😍

File: src/Akeneo/Platform/Job/back/Infrastructure/Query/SearchJobExecution.php [hunk 3]
```diff
@@ -72,6 +72,9 @@ public function search(SearchJobExecutionQuery $query): array
         $queryParamsTypes = $this->buildQueryParamsTypes();
 
         $page = $query->page;
+        if (!$query->hasOneFilterSet() && $page > 50) {
```
Comment (juliensnz) [inline src/Akeneo/Platform/Job/back/Infrastructure/Query/SearchJobExecution.php:75]: maybe extract this as a const? It' a bonus, not mandatory :)

File: src/Akeneo/Platform/Job/back/Infrastructure/Query/SearchJobExecution.php [hunk 4]
```diff
@@ -72,6 +72,9 @@ public function search(SearchJobExecutionQuery $query): array
...
+            throw new \InvalidArgumentException('The page number can not be greater than 50 when no filter are set');
```
Comment (juliensnz) [inline src/Akeneo/Platform/Job/back/Infrastructure/Query/SearchJobExecution.php:76]: I don't understand this: when some filter are set, we should neither go above 50 no?

File: src/Akeneo/Platform/Job/back/tests/.php_cd.php (hunk 2 above)
Comment (juliensnz) [reply src/Akeneo/Platform/Job/back/tests/.php_cd.php:30]: 🎉

**Review by juliensnz — APPROVED** [review]

File: src/Akeneo/Platform/Job/back/Infrastructure/Query/SearchJobExecution.php (hunk 3 above)
Comment (ghoussard) [reply src/Akeneo/Platform/Job/back/Infrastructure/Query/SearchJobExecution.php:75]: 👍🏻 Yep, it will be done in RAC-1101

File: src/Akeneo/Platform/Job/back/Infrastructure/Query/SearchJobExecution.php (hunk 4 above)
Comment (ghoussard) [reply src/Akeneo/Platform/Job/back/Infrastructure/Query/SearchJobExecution.php:76]: It's weird for me too but this is the behavior expected by Félix : user could go above 50 if he use at leat one filter (specified in jira issue comment and confirmed by it this morning) => "With search and filters, all jobs available"

File: front-packages/akeneo-design-system/src/components/Helper/Helper.stories.mdx [hunk 5]
```diff
@@ -124,6 +124,30 @@ non fermentum diam nisl sit amet erat. `}
   </Story>
 </Canvas>
 
+## Sticky
+
+<Canvas>
+  <Story name="Sticky">
```
Comment (ValentinMumble) [inline front-packages/akeneo-design-system/src/components/Helper/Helper.stories.mdx:130]: Maybe we can take the SectionTitle as an example here https://dsm.akeneo.com/15845/?path=/docs/components-section-title--standard#sticky
Because the story looks weird as is 
image
What do you think?

File: front-packages/akeneo-design-system/src/components/Helper/Helper.stories.mdx [hunk 6]
```diff
@@ -124,6 +124,30 @@ non fermentum diam nisl sit amet erat. `}
...
+            This is a sticky helper.
+          </Helper>
+          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed non risus.Suspendisse lectus tortor,
+          dignissim sit amet, adipiscing nec, ultricies sed, dolor. Cras elementum ultrices diam.
+          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed non risus.Suspendisse lectus tortor,
+          dignissim sit amet, adipiscing nec, ultricies sed, dolor. Cras elementum ultrices diam.
+          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed non risus.Suspendisse lectus tortor,
+          dignissim sit amet, adipiscing nec, ultricies sed, dolor. Cras elementum ultrices diam.
+          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed non risus.Suspendisse lectus tortor,
+          dignissim sit amet, adipiscing nec, ultricies sed, dolor. Cras elementum ultrices diam.
```
Comment (ValentinMumble) [inline front-packages/akeneo-design-system/src/components/Helper/Helper.stories.mdx:144]:
```suggestion
Logoden biniou degemer mat an penn ar bed trizek, mall Mikael Entraven paotrig Melwenn kegin lizherenn warc’hoazh meur, morzhol fourchetez señin gazek pleg gwalc’hiñ pediñ. Verc’h boull pignat doñv vro kilañ miliner gwengolo Santeg sioul vi moc’h Plelann-Vihan mamm, poent enebour enep seitek Pornizhan warc’hoazh derc’hent izel gaou gwerenn huanadiñ. Doare linenn paotrig matezh speredek plijadur wern piler eviti, torchañ neuze Landreger Pont’n-Abad vunutenn se gwirionez ganit munutenn, disheol naontek askell Argol Tregastell pomper ger.
```

Comment (ValentinMumble) [reply front-packages/akeneo-design-system/src/components/Helper/Helper.stories.mdx:144]: It's a good practice to use https://www.logoden-biniou.com as text sample generator

File: src/Akeneo/Platform/Job/front/process-tracker/src/feature/pages/JobExecutionList.test.tsx [hunk 7]
```diff
@@ -187,3 +216,40 @@ test('it can change page', () => {
...
+  mockedSize = 1;
+  mockUseManyRows = true;
+
+  renderWithProviders(<JobExecutionList />);
+
+  userEvent.click(screen.getByLabelText('akeneo_job_process_tracker.user_filter.label:'));
+  userEvent.click(screen.getAllByText('admin')[screen.getAllByText('admin').length - 1]);
+
+  for (let i = 2; i <= 50; i++) {
+    userEvent.click(screen.getByTitle(`No. ${i}`));
```
Comment (ValentinMumble) [inline src/Akeneo/Platform/Job/front/process-tracker/src/feature/pages/JobExecutionList.test.tsx:250]: 🤔  Sad that this is the only way to reach page 50....

**Review by ValentinMumble — APPROVED** [review]

File: front-packages/akeneo-design-system/src/components/Helper/Helper.stories.mdx (hunk 5 above)
Comment (ghoussard) [reply front-packages/akeneo-design-system/src/components/Helper/Helper.stories.mdx:130]: Good idea 👍🏻 I will try that

File: front-packages/akeneo-design-system/src/components/Helper/Helper.stories.mdx (hunk 6 above)
Comment (StevenVAIDIE) [reply front-packages/akeneo-design-system/src/components/Helper/Helper.stories.mdx:144]: bretagne

Comment (ghoussard) [reply front-packages/akeneo-design-system/src/components/Helper/Helper.stories.mdx:144]: Don't rely me to put breizh shit in my code 💩

File: src/Akeneo/Platform/Job/front/process-tracker/src/feature/pages/JobExecutionList.test.tsx (hunk 7 above)
Comment (ghoussard) [reply src/Akeneo/Platform/Job/front/process-tracker/src/feature/pages/JobExecutionList.test.tsx:250]: cc @StevenVAIDIE

File: src/Akeneo/Platform/Job/back/Infrastructure/Query/SearchJobExecution.php (hunk 4 above)
Comment (StevenVAIDIE) [reply src/Akeneo/Platform/Job/back/Infrastructure/Query/SearchJobExecution.php:76]: We should rediscuss about it because this is not was we decided in grooming

File: src/Akeneo/Platform/Job/back/Infrastructure/Symfony/Resources/translations/jsmessages.en_US.yml [hunk 8]
```diff
@@ -36,8 +47,10 @@ akeneo_job_process_tracker:
     abandoned: Abandoned
     completed: Completed
     failed: Failed
-    started: Started
     starting: Starting
     stopped: Stopped
     stopping: Stopping
     unknown: Unknown
+    in_progress: In progress
```
Comment (StevenVAIDIE) [inline src/Akeneo/Platform/Job/back/Infrastructure/Symfony/Resources/translations/jsmessages.en_US.yml:54]: We cannot merge the two translation keys ?

File: src/Akeneo/Platform/Job/front/process-tracker/src/feature/pages/JobExecutionList.test.tsx (hunk 7 above)
Comment (ghoussard) [reply src/Akeneo/Platform/Job/front/process-tracker/src/feature/pages/JobExecutionList.test.tsx:250]: Maybe we can pass by the mock of useStoredJobExecutionFilter to have directly the page 50, I will try it 😄

Comment (StevenVAIDIE) [reply src/Akeneo/Platform/Job/front/process-tracker/src/feature/pages/JobExecutionList.test.tsx:250]: The last page is not accessible ? Normally it should:

Screenshot from 2021-12-03 17-51-43

File: src/Akeneo/Platform/Job/front/process-tracker/src/feature/pages/JobExecutionList.tsx [hunk 9]
```diff
@@ -11,16 +11,25 @@ import {
...
+const MAX_PAGE_WITHOUT_FILTER = 50;
+
 const JobExecutionList = () => {
   const activityHref = useRoute('pim_dashboard_index');
   const translate = useTranslate();
 
   const [jobExecutionFilter, setJobExecutionFilter] = useStoredJobExecutionFilter();
   const [jobExecutionTable, refreshJobExecutionTable] = useJobExecutionTable(jobExecutionFilter);
-  const matchesCount = jobExecutionTable === null ? 0 : jobExecutionTable.matches_count;
+  const matchesCount =
```
Comment (StevenVAIDIE) [inline src/Akeneo/Platform/Job/front/process-tracker/src/feature/pages/JobExecutionList.tsx:25]: You should not override it here => the matches count displayed at the top of the page should be the right number

File: src/Akeneo/Platform/Job/front/process-tracker/src/feature/pages/JobExecutionList.tsx [hunk 10]
```diff
@@ -84,15 +93,20 @@ const JobExecutionList = () => {
...
+                  <Helper level="warning" sticky={44}>
+                    {translate('akeneo_job_process_tracker.max_page_without_filter_helper')}
+                  </Helper>
+                )}
                 <Pagination
-                  sticky={44}
+                  sticky={displayPaginationWarning ? 88 : 44}
                   itemsPerPage={jobExecutionFilter.size}
                   currentPage={jobExecutionFilter.page}
                   totalItems={matchesCount}
```
Comment (StevenVAIDIE) [inline src/Akeneo/Platform/Job/front/process-tracker/src/feature/pages/JobExecutionList.tsx:105]:
```suggestion
                  totalItems={Math.max(matchesCount, MAX_PAGE_WITHOUT_FILTER * jobExecutionFilter.size}
```

File: src/Akeneo/Platform/Job/back/Infrastructure/Query/SearchJobExecution.php (hunk 4 above)
Comment (StevenVAIDIE) [reply src/Akeneo/Platform/Job/back/Infrastructure/Query/SearchJobExecution.php:76]: For exemple the user can silter by status unknown, the user will not be restricted

**Review by StevenVAIDIE — APPROVED** [review]
We should discuss with Felix about it

File: src/Akeneo/Platform/Job/front/process-tracker/src/feature/pages/JobExecutionList.test.tsx (hunk 7 above)
Comment (ghoussard) [reply src/Akeneo/Platform/Job/front/process-tracker/src/feature/pages/JobExecutionList.test.tsx:250]: Yep but here, the last page is 51, not 50. We can click on 51 and then click on 50 ?

File: front-packages/akeneo-design-system/src/components/Helper/Helper.stories.mdx [hunk 11]
```diff
@@ -130,11 +131,11 @@ non fermentum diam nisl sit amet erat. `}
   <Story name="Sticky">
     {args => {
       return (
-        <Scrollable height={100}>
-          <Helper {...args} level="info" sticky={44}>
+        <Scrollable height={300}>
+          <Helper level="warning" sticky={0}>
             This is a sticky helper.
           </Helper>
-Logoden biniou degemer mat an penn ar bed trizek, mall Mikael Entraven paotrig Melwenn kegin lizherenn warc’hoazh meur, morzhol fourchetez señin gazek pleg gwalc’hiñ pediñ. Verc’h boull pignat doñv vro kilañ miliner gwengolo Santeg sioul vi moc’h Plelann-Vihan mamm, poent enebour enep seitek Pornizhan warc’hoazh derc’hent izel gaou gwerenn huanadiñ. Doare linenn paotrig matezh speredek plijadur wern piler eviti, torchañ neuze Landreger Pont’n-Abad vunutenn se gwirionez ganit munutenn, disheol naontek askell Argol Tregastell pomper ger.
```
Comment (ValentinMumble) [inline front-packages/akeneo-design-system/src/components/Helper/Helper.stories.mdx]: https://emoji.slack-edge.com/T031L1UKF/sad-dog/ac79218e153de6de.jpg

Comment (ghoussard) [reply front-packages/akeneo-design-system/src/components/Helper/Helper.stories.mdx]: 😢

File: src/Akeneo/Platform/Job/back/Infrastructure/Symfony/Resources/translations/jsmessages.en_US.yml (hunk 8 above)
Comment (ghoussard) [reply src/Akeneo/Platform/Job/back/Infrastructure/Symfony/Resources/translations/jsmessages.en_US.yml:54]: As I created a new object that contains IN_PROGRESS instead of STARTED, I think it is clearer to replace it also in translations
